### PR TITLE
feat: Added support for custom domain

### PR DESCRIPTION
**Pull request by ngarg-mparticle**
## Summary
 - Added support for custom domain. Airship documentation: https://docs.airship.com/reference/integration/custom-domain-proxy/?groupid=android-kotlin

### CODE REVIEW

File: src/main/kotlin/com/mparticle/kits/UrbanAirshipConfiguration.kt [hunk 1]
```diff
@@ -67,6 +68,7 @@ class UrbanAirshipConfiguration(settings: Map<String, String>) {
         private const val KEY_APP_KEY = "applicationKey"
         private const val KEY_APP_SECRET = "applicationSecret"
         private const val KEY_DOMAIN = "domain"
+        private const val KEY_INITIAL_CONFIG_URL = "initialConfigUrl"
```
Comment (jasonmgeorge) [inline src/main/kotlin/com/mparticle/kits/UrbanAirshipConfiguration.kt:71]: Configuration setting key will be added to align with UI description.

```suggestion
        private const val KEY_CUSTOM_DOMAIN_PROXY_URL = "customDomainProxyUrl"
```

File: src/main/kotlin/com/mparticle/kits/UrbanAirshipConfiguration.kt [hunk 2]
```diff
@@ -95,6 +97,9 @@ class UrbanAirshipConfiguration(settings: Map<String, String>) {
         applicationKey = settings[KEY_APP_KEY]
         applicationSecret = settings[KEY_APP_SECRET]
         domain = settings[KEY_DOMAIN]
+        if (settings.containsKey(KEY_INITIAL_CONFIG_URL)) {
+            initialConfigUrl = settings[KEY_INITIAL_CONFIG_URL]
+        }
```
Comment (jasonmgeorge) [inline src/main/kotlin/com/mparticle/kits/UrbanAirshipConfiguration.kt:102]: Based on the configuration setting key change..

```suggestion
        if (settings.containsKey(KEY_CUSTOM_DOMAIN_PROXY_URL)) {
            initialConfigUrl = settings[KEY_CUSTOM_DOMAIN_PROXY_URL]
        }
```

**Comment by glyn-leigh** [comment]
This is Glynis from the Sam's Club Platform Team! An overview on the testing that we conducted:
We made sure it would compile, and we changed the gradle file a bit to be able to use it as a library for our project.
We used the mParticle dashboard to put in our custom domain, and then debugged the kit code to ensure that:
1. The custom domain was being passed through
2. We checked the network traffic to ensure it was being passed through for push and in-app notifications
3. Airship was still functioning normally
Things look good from our end, so we're eagerly awaiting the official SDK update.

**Comment by rmi22186** [comment]
@glyn-leigh - thanks for your support with this!  i've merged and this will go out with our next android release

**Comment by glyn-leigh** [comment]
@rmi22186 
Do you know what date that release is slated for?

**Comment by Mansi-mParticle** [comment]
@glyn-leigh this has been released, you can check at https://repo1.maven.org/maven2/com/mparticle/android-urbanairship-kit/5.55.5/

**Pull request merged**